vita3k/modules/SceHttp/SceHttp.cpp

```diff
@@ -303,35 +1045,153 @@ EXPORT(int, sceHttpSslIsCtxCreated) {
 }
 
 EXPORT(int, sceHttpTerm) {
-    return UNIMPLEMENTED();
+    // clear everything
+
+    for (auto &tmpl : host.http.templates) {
+        CALL_EXPORT(sceHttpDeleteTemplate, tmpl.first);
+    }
+    host.http.templates.clear();
+
+    for (auto &conn : host.http.connections) {
+        CALL_EXPORT(sceHttpDeleteConnection, conn.first);
+    }
+    host.http.connections.clear();
+
+    for (auto &req : host.http.requests) {
+        CALL_EXPORT(sceHttpDeleteRequest, req.first);
+    }
+    host.http.requests.clear();
+
+    return 0;
 }
 
 EXPORT(int, sceHttpUnsetEpoll) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceHttpUriBuild) {
+EXPORT(int, sceHttpUriBuild, char *out, unsigned int *require, unsigned int prepare, const SceHttpUriElement *srcElement, unsigned int option) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (!out || !srcElement)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_VALUE);
+
+    // TODO
+
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceHttpUriEscape) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceHttpUriEscape, char *out, unsigned int *require, unsigned int prepare, const char *in) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (!out || !in)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_VALUE);
+
+    // TODO: test
+
+    auto encoded = net_utils::encodeURIComponent(std::string(in)).c_str();
```

Comment:
The string containing the car array that encoded is pointing to will be destroyed at the end of the line. Therefore encoded will point to undefined content when it is copied next line.

Reply:
Same issue is present (dangling pointer)